sql/item_timefunc.cc

```diff
@@ -1212,28 +1212,19 @@ longlong Item_func_year::val_int_endpoint(bool left_endp, bool *incl_endp)
 }
 
 
-bool Item_func_unix_timestamp::get_timestamp_value(my_time_t *seconds,
+bool Item_func_unix_timestamp::get_timestamp_value(longlong *seconds,
```

Comment:
I dont think a change like this is a good idea.
I may see two points:
1. there may be some storage format tied to `my_time_t` size;
2. some other conversion issues

Other than that I don't see why not just redefine `my_time_t` to something larger.

Anyways, neither I do think that we'll not move to 128-bit architecture to 2038.
At least, making a 128-bit cache line is rather a demand.
With this I wouldn't think that `long long` will remain 64 bit, especially with guarantee. Most likely, it will depend on the implementation, just like now.

So at least you should use something constantly-sized, if you want a strictly 64-bit large output.
`uint64_t` is good enough (I think we support `<stdint.h>`).

I would also think of making a dedicated type like `my_time_t_64` or something, to which we will eventlually move the rest of the code base.

Reply:
> I dont think a change like this is a good idea. I may see two points:
> 
>     1. there may be some storage format tied to `my_time_t` size;
That function is only called from within Item_func_unix_tijmestamp itself, it has no relation to the storage format.

>     2. some other conversion issues
I am still working on verifying and correcting all tests. Can you be more specific as to which conversion issues you are expecting?

> Other than that I don't see why not just redefine `my_time_t` to something larger.

Redefining my_time_t to something else has very large implications, notably it means a header change for 3rd-party software. But also tons of other MySQL internals will be affected (including the storage format) which I don't think is a good idea.
 
> I would also think of making a dedicated type like `my_time_t_64` or something, to which we would eventlually move the rest of the code base.

Not a bad idea, though if you look carefully you'll see that all I really did was replace the internal conversion done by UNIX_TIMESTAMP() from TIMESTAMP to DATETIME and enable FROM_UNIXTIME() parsing of dates > 2038. I tried hard to make sure nothing else would be affected, especially the storage format of TIMESTAMP. DATETIME columns themselves are fine, they already support dates from 0000 to 9999 if I recall correctly.

Reply:
A 128-bit (16-byte) cache line would be extraordinarily small, I think. IA-32 and AMD64 implementations typically have 64-byte (512-bit) cache lines. On other ISAs such as ARMv8 or s390x, the cache line can be even larger.

Reply:
I don't see how cache lines are related to a function argument?

Reply:
Maybe @FooBarrior was thinking about the alignment of the pointer target? Based on my experience from [MDEV-28091](https://jira.mariadb.org/browse/MDEV-28091), normal `malloc()` might already return 128-bit aligned memory on 64-bit systems. I believe that 64-bit alignment is guaranteed already on 32-bit systems.

Reply:
The pointer argument is the return value for this function. The function itself returns true/false indicating success/failure. Since the caller asked for the timestamp value, this value has to be returned somehow.

@FooBarrior was referring to `longlong` being possibly larger than 64-bits on 128-bit architectures and suggested using `(u)int64_t` instead to define the exact size for `seconds`. This suggestion is valid and correct.